packages/core/strapi/src/services/document-service/index.ts

```diff
@@ -0,0 +1,126 @@
+import type { Strapi, DocumentService, EntityValidator, EventHub } from '@strapi/types';
+import { convertQueryParams } from '@strapi/utils';
+import type { Database } from '@strapi/database';
+
+import uploadFiles from '../utils/upload-files';
+import { pickSelectionParams } from './params';
+
+const { transformParamsToQuery } = convertQueryParams;
+
+/**
+ * TODO: TESTS
+ * TODO: Components
+ * TODO: Entity Validation
+ * TODO: Lifecycles
+ *        Plugin extensions
+ * TODO: D&P
+ * TODO: i18n
+ * TODO: Apply default parameters (status & locale)
+ * TODO: Sanitization
+ * TODO: Other methods
+ *        Count
+ *        FindPage
+ *        Clone
+ *        Load
+ *        LoadPages
+ *        DeleteMany
+ * TODO: Webhooks
+ * TODO: Audit logs
+ * TODO: File upload
+ * TODO: Transactions?
+ */
+
+const createDocumentService = ({
+  strapi,
+  db,
+}: {
+  strapi: Strapi;
+  db: Database;
+  eventHub: EventHub;
+  entityValidator: EntityValidator;
+}): DocumentService.DocumentService => ({
+  uploadFiles,
+
+  async findMany(uid, params) {
+    const { kind } = strapi.getModel(uid);
+
+    const query = transformParamsToQuery(uid, params || ({} as any));
+
+    if (kind === 'singleType') {
+      return db.query(uid).findOne(query) as any;
+    }
```

Comment:
Do we want to keep this? @alexandrebodin 
I am not sure if that was decided, but probably removing it would make the migration a bit harder for users